### ISSUE TITLE
Log flood due to "Botch on admin job for admin : already in use!!"

### ISSUE THREAD

**Issue by guestisp** [backuppc/backuppc#303]
v4.3.1

When nighly job spawn multiple days (in my case, even 1 week), log are
flooded with:

Botch on admin job for  admin : already in use!!

would be possible to relax this log or add a sort of ratelimit like
syslog does ?

There is no need to log the same error line 50 times per second, like
the following:

2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!
2019-12-26 16:36:41 Botch on admin job for  admin : already in use!!

**Comment by craigbarratt** [comment]
Sorry about the delay in replying - I've been traveling for the last 3 weeks with no internet access for a good part of that time.

As a quick workaround to avoid the flooding, immediately after the Botch message in bin/BackupPC, comment out the unshift line:
```
print(LOG $bpc->timeStamp,
           "Botch on admin job for $host: already in use!!\n");
#
# Just put this request back on the end of the queue.
#
### unshift(@CmdQueue, $req);  <- commented out this line
return
```
Let me look at the BackupPC_nightly code to figure out why it queues a new run before the old one has finished.

BTW, what are your settings for:
```
$Conf{MaxBackupPCNightlyJobs}
$Conf{BackupPCNightlyPeriod}
$Conf{PoolSizeNightlyUpdatePeriod}
```

**Comment by craigbarratt** [comment]
I pushed a fix 7626b33 for both issues - correctly checking whether BackupPC_nightly is still running before queuing another, and dropping a command if it is already running.  However, I haven't tested it yet.

**Comment by guestisp** [comment]
> $Conf{MaxBackupPCNightlyJobs}
> $Conf{BackupPCNightlyPeriod}
> $Conf{PoolSizeNightlyUpdatePeriod}

```
$Conf{MaxBackupPCNightlyJobs} = 2;
$Conf{BackupPCNightlyPeriod} = 32;
$Conf{PoolSizeNightlyUpdatePeriod} = 128;
```
